### DISCUSSION TITLE
Introducing An All-New GUI & Website

### DISCUSSION THREAD

**Discussion by VerisimilitudeX** [VerisimilitudeX/DNAnalyzer#310] [Announcements]
# GUI
Thanks, @frankschmitt for implementing a cross-platform GUI application for DNAnalyzer. Currently, it supports the following features:
- Set the name of the DNA file to analyze
- Set minimum and maximum number of reading frames
- Run analysis

The Met CLI continues to be the best way to run the program, with more features to be added to both in the future.

# Website
Thanks, @krishnabiradar310802 for developing a website that creates awareness for DNAnalyzer and outlines the main features. It is accessible via https://www.dnanalyzer.live/ and is hosted on GitHub Pages. We are currently working on linking the backend to the frontend with the help of @aryanagrawal8808. @zackwebster and @BLUE-DEVIL1134 are working on improving the frontend code.

As always, thanks, everyone for your contributions! I'll see you all in the next release.

## What's Changed
* Fixed bugs in README by @Verisimilitude11 in https://github.com/Verisimilitude11/DNAnalyzer/pull/201
* Format code with autopep8 and google-java-format by @deepsource-autofix in https://github.com/Verisimilitude11/DNAnalyzer/pull/200
* Webpage  by @krishnabiradar310802 in https://github.com/Verisimilitude11/DNAnalyzer/pull/202
* Updated padding  by @krishnabiradar310802 in https://github.com/Verisimilitude11/DNAnalyzer/pull/203
* Refactor CodonDataConstants by @Kit-p in https://github.com/Verisimilitude11/DNAnalyzer/pull/194
* Revert "Refactor CodonDataConstants" by @Verisimilitude11 in https://github.com/Verisimilitude11/DNAnalyzer/pull/209
* Refactor CmdArgs.run() by @ishche in https://github.com/Verisimilitude11/DNAnalyzer/pull/196
* Format code with autopep8 and google-java-format by @deepsource-autofix in https://github.com/Verisimilitude11/DNAnalyzer/pull/211
* Organize code into folders by @plopez26DPU in https://github.com/Verisimilitude11/DNAnalyzer/pull/214
* Add Wiki markdown by @plopez26DPU in https://github.com/Verisimilitude11/DNAnalyzer/pull/217
* Refactor CmdArgs by @singloon in https://github.com/Verisimilitude11/DNAnalyzer/pull/198
* Refactor Properties.isRandomDNA(String) - Issue #73 by @B4CKF1SH in https://github.com/Verisimilitude11/DNAnalyzer/pull/97
* Fix deploy GitHub page workflow by @Kit-p in https://github.com/Verisimilitude11/DNAnalyzer/pull/227
* Changing wording, Gradle may not work with JDK 19. by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/235
* Update CodonFrame.java by @sourabhkashyap95 in https://github.com/Verisimilitude11/DNAnalyzer/pull/213
* refactor: protein finder by @boaglio in https://github.com/Verisimilitude11/DNAnalyzer/pull/216
* Format code with autopep8 and google-java-format by @deepsource-autofix in https://github.com/Verisimilitude11/DNAnalyzer/pull/221
* Refactor CodonDataConstants by @LoicB in https://github.com/Verisimilitude11/DNAnalyzer/pull/233
* Refactor output codons by @plopez26DPU in https://github.com/Verisimilitude11/DNAnalyzer/pull/234
* Format code with autopep8 and google-java-format by @deepsource-autofix in https://github.com/Verisimilitude11/DNAnalyzer/pull/238
* About us page by @krishnabiradar310802 in https://github.com/Verisimilitude11/DNAnalyzer/pull/241
* New CLI documentation by @plopez26DPU in https://github.com/Verisimilitude11/DNAnalyzer/pull/244
* Added some new future improvements to the read.md by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/257
* Update .deepsource.toml by @deepsource-autofix in https://github.com/Verisimilitude11/DNAnalyzer/pull/259
* fixed typo in link by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/264
* Added Dependabot Configuration by @jenistenxavier in https://github.com/Verisimilitude11/DNAnalyzer/pull/263
* Refactored the methods relating to printing neucleotide count by @zaino97 in https://github.com/Verisimilitude11/DNAnalyzer/pull/253
* Use const declarations for variables that are never reassigned by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/267
* Count base pairs update by @garyglatfelter in https://github.com/Verisimilitude11/DNAnalyzer/pull/260
* Refactorings of getAminoAcidMapping by @Speedro in https://github.com/Verisimilitude11/DNAnalyzer/pull/243
* [ImgBot] Optimize images by @imgbot in https://github.com/Verisimilitude11/DNAnalyzer/pull/254
* Fix For Broken CodeQL Java Analysis by @Speedro in https://github.com/Verisimilitude11/DNAnalyzer/pull/272
* Update .deepsource.toml by @deepsource-autofix in https://github.com/Verisimilitude11/DNAnalyzer/pull/279
* Adding information about DNA to README by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/282
* Add .DS_Store to .gitignore by @ImpossibleReality in https://github.com/Verisimilitude11/DNAnalyzer/pull/288
* 'toggle' is never reassigned. Use 'const' instead by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/281
* Condensed code in countBasePairs as to resolve #274 by @garyglatfelter in https://github.com/Verisimilitude11/DNAnalyzer/pull/285
* Issue 252 publish gradle package by @jaybob007 in https://github.com/Verisimilitude11/DNAnalyzer/pull/290
* #270 Added highCoverageRegions call to CmdArgs by @jaybob007 in https://github.com/Verisimilitude11/DNAnalyzer/pull/284
* Refactorings of DNAAnalysis.countBasePairs(String) by @Speedro in https://github.com/Verisimilitude11/DNAnalyzer/pull/278
* Undocumented declaration found by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/292
* Refactoring #276 by @salahhalkhoums in https://github.com/Verisimilitude11/DNAnalyzer/pull/287
* Feature/add javafx gui #121 by @frankschmitt in https://github.com/Verisimilitude11/DNAnalyzer/pull/286
* Code refactoring by @ArchontisKostis in https://github.com/Verisimilitude11/DNAnalyzer/pull/219
* Refactor ProteinFinder.getProtein method by @Ravina-Deogadkar in https://github.com/Verisimilitude11/DNAnalyzer/pull/294
* Updating correct wording for feature by @LimesKey in https://github.com/Verisimilitude11/DNAnalyzer/pull/304

## New Contributors
* @krishnabiradar310802 made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/202
* @ishche made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/196
* @plopez26DPU made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/214
* @singloon made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/198
* @LimesKey made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/235
* @sourabhkashyap95 made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/213
* @boaglio made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/216
* @LoicB made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/233
* @jenistenxavier made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/263
* @zaino97 made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/253
* @garyglatfelter made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/260
* @ImpossibleReality made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/288
* @jaybob007 made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/290
* @salahhalkhoums made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/287
* @ArchontisKostis made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/219
* @Ravina-Deogadkar made their first contribution in https://github.com/Verisimilitude11/DNAnalyzer/pull/294

**Full Changelog**: https://github.com/Verisimilitude11/DNAnalyzer/compare/v2.0.0...v2.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Verisimilitude11/DNAnalyzer/releases/tag/v2.1.0'>Introducing An All-New GUI & Website</a>.</em>

**Comment by krishnabiradar310802** [comment]
It was great working with you @Verisimilitude11.Hope we work together in future.

**Reply by VerisimilitudeX** [reply]
Yep, thanks to you too! If you get some time soon, could you please help us make a navbar? Thanks!

**Reply by krishnabiradar310802** [reply]
Yes we can discuss on this.

**Reply by IcebergSnow** [reply]
Very Cool 👍

**Reply by VerisimilitudeX** [reply]
Thanks @IcebergSnow!

**Comment by IcebergSnow** [comment]
Awesome!🎉

**Reply by VerisimilitudeX** [reply]
Thanks for your support!! 🎉
